driver/src/main/java/org/kaazing/k3po/driver/internal/control/handler/ControlServerHandler.java

```diff
@@ -105,6 +109,29 @@ public void prepareReceived(final ChannelHandlerContext ctx, MessageEvent evt) t
             return;
         }
 
+        if (lastRobot != null && ! lastRobot.getDisposedFuture().isDone()) {
+            // we need to wait for it to end
+            lastRobot.getDisposedFuture().addListener(new ChannelFutureListener() {
+                @Override
+                public void operationComplete(ChannelFuture future) throws Exception {
```

Comment:
This is ok, but it might be cleaner and more future proof to write it without a dependency on the underlying thread model.  I.E. solve for the case where there are more than one thread